podcasts/AutoplayHelper.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+class AutoplayHelper {
+    static let shared = AutoplayHelper()
+
+    private let userDefaults: UserDefaults
+    private let userDefaultsKey = "playlist"
+
+    var lastPlaylist: EpisodesDataManager.Playlist? {
+        if let rawDictionary = userDefaults.dictionary(forKey: userDefaultsKey),
+           let dictData = try? JSONSerialization.data(withJSONObject: rawDictionary) {
+            return try? JSONDecoder().decode(EpisodesDataManager.Playlist.self, from: dictData)
+        }
+
+        return nil
+    }
+
+    init(userDefaults: UserDefaults = UserDefaults.standard) {
+        self.userDefaults = userDefaults
+    }
+
+    func playedAt(playlist: EpisodesDataManager.Playlist?) {
+        #if !os(watchOS)
+        save(selectedPlaylist: playlist)
+        #endif
+    }
+
+    private func save(selectedPlaylist playlist: EpisodesDataManager.Playlist?) {
+        guard let playlist else {
+            userDefaults.removeObject(forKey: userDefaultsKey)
+            return
+        }
+
+        if let data = try? JSONEncoder().encode(playlist),
+           let dict = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) {
+            userDefaults.set(dict, forKey: userDefaultsKey)
+        }
```

Comment:
Is there a reason we're converting the encoded data into a dictionary before saving it when we could just save the data directly?


```suggestion
        guard let data = try? JSONEncoder().encode(playlist) else {
            return
        }

        userDefaults.set(data, forKey: userDefaultsKey)
```